### PR TITLE
feat: Improve Logging in Dynamo

### PR DESCRIPTION
**Pull request by peri044**
# Description

* Move compile_module from backends to top level since it is used by export and compile.
* Use logging library in `torch_tensorrt/_compile.py`. The original logging.py is using C++ utilities which probably isn't needed for this file.
* Define a root level logger with namespace `torch_tensorrt.dynamo` defined in `torch_tensorrt/dynamo/__init__.py`. Setting debug=True now invokes all the module level loggers to follow this level since they are children of the root level logger. This fixes partitioning log in the cases where it doesn't get logged sometimes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/aten_tracer.py [hunk 1]
```diff
@@ -85,7 +87,6 @@ def setting_python_recursive_limit(limit: int = 10000) -> Generator[None, None,
         sys.setrecursionlimit(default)
 
 
-@req_torch_version("2.dev")
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/aten_tracer.py:88]: These function wrappers are intended to warn users compiling with the legacy library that these functions will not work with Torch 1.13. Are they removed here to be replaced by alternative log messaging?

**Review by gs-olive — COMMENTED** [review]
Unification of the `compile_module` function and the logging bugfix look great! Looks like one of the ts tests has a logging conflict since it expects the C++ version, so the import could be specialized more for that test case. Left a comment regarding version requirements for legacy.

File: py/torch_tensorrt/dynamo/aten_tracer.py (hunk 1 above)
Comment (peri044) [reply py/torch_tensorrt/dynamo/aten_tracer.py:88]: So mypy complains about these decorators. I've changed the decorators according to mypy conventions. This should be resolved now.

File: py/torch_tensorrt/dynamo/utils.py [hunk 2]
```diff
@@ -169,7 +174,7 @@ def parse_dynamo_kwargs(kwargs: Any) -> CompilationSettings:
     return settings
 
 
-def req_torch_version(min_torch_version: str = "2.dev") -> Callable[..., Any]:
+def req_torch_version(func: Callable[P, T]) -> Callable[P, T]:
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/utils.py:177]: Why was this changed?

Comment (narendasan) [reply py/torch_tensorrt/dynamo/utils.py:177]: The arguments dont seem right

Comment (peri044) [reply py/torch_tensorrt/dynamo/utils.py]: reverted back. Seems like mypy passes now with no changes. Although I'm not sure why it was giving issues before
